internal/serve/serve.go

```diff
@@ -155,6 +167,47 @@ func initHandlerDeps(cfg Configs) (handlerDeps, error) {
 	}
 	go ensureChannelAccounts(channelAccountService, int64(cfg.NumberOfChannelAccounts))
 
+	// TSS
+	txServiceOpts := tssservices.TransactionServiceOptions{
+		DistributionAccountSignatureClient: cfg.DistributionAccountSignatureClient,
+		ChannelAccountSignatureClient:      cfg.ChannelAccountSignatureClient,
+		HorizonClient:                      &horizonClient,
+		BaseFee:                            int64(cfg.BaseFee), // Reuse horizon base fee for RPC??
+	}
+	tssTxService, err := tssservices.NewTransactionService(txServiceOpts)
+	if err != nil {
+		return handlerDeps{}, fmt.Errorf("instantiating tss transaction service: %w", err)
+	}
+	httpClient := http.Client{Timeout: time.Duration(30 * time.Second)}
+	rpcService, err := services.NewRPCService(cfg.RPCURL, &httpClient)
+	if err != nil {
+		return handlerDeps{}, fmt.Errorf("instantiating rpc service: %w", err)
+	}
+
+	// re-use same context as above??
+	store := tssstore.NewStore(dbConnectionPool)
+	txManager := tssservices.NewTransactionManager(tssservices.TransactionManagerConfigs{
+		TxService:  tssTxService,
+		RPCService: rpcService,
+		Store:      store,
+	})
+	tssChannelConfigs := tsschannel.RPCCallerChannelConfigs{
+		TxManager:     txManager,
+		Store:         store,
+		MaxBufferSize: cfg.RPCCallerServiceChannelBufferSize,
+		MaxWorkers:    cfg.RPCCallerServiceChannelMaxWorkers,
+	}
+	rpcCallerServiceChannel := tsschannel.NewRPCCallerChannel(tssChannelConfigs)
+
+	router := tssrouter.NewRouter(tssrouter.RouterConfigs{
```

Comment:
It's a good idea to check in the `NewRouter` and `NewStore` functions that the parameters are passed in correctly and are not `nil`. If a parameter is invalid, they can return an error.

Reply:
I did add a nil check for NewStore, but kept NewRouter the way it is right now, for reasons I explained in another comment, I would like the option of some of these channels being nil